### PR TITLE
Persian Language support

### PR DESCRIPTION
**Pull request by omidp**

### CODE REVIEW

**Comment by wjur** [comment]
I'm sorry to write back so late, @omidp .
Could you update the PR (merge/rebase to master) and fix emerging issues, please?
You can run `gradlew check` to be sure that the checks pass.

I look forward merging your contribution.

File: src/main/java/pl/allegro/finance/tradukisto/internal/Container.java [hunk 1]
```diff
@@ -157,6 +158,22 @@ public static Container brazilianPortugueseContainer() {
...
         return new Container(converter, bigDecimalBankingMoneyValueConverter);
     }
+    
+    public static Container persianContainer()
+    {
+        PersianValues persianValues = new PersianValues();
+        HundredsToWordsConverter hundredsToStringConverter = new HundredsToWordsConverter(persianValues.baseNumbers(),
+                persianValues.twoDigitsNumberSeparator());
+
+        pl.allegro.finance.tradukisto.internal.languages.persian.IntegerToWordsConverter integerConverter =
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/Container.java:168]: please use import and not the full qualified class name

File: src/main/java/pl/allegro/finance/tradukisto/internal/Container.java [hunk 2]
```diff
@@ -157,6 +158,22 @@ public static Container brazilianPortugueseContainer() {
...
+        PersianValues persianValues = new PersianValues();
+        HundredsToWordsConverter hundredsToStringConverter = new HundredsToWordsConverter(persianValues.baseNumbers(),
+                persianValues.twoDigitsNumberSeparator());
+
+        pl.allegro.finance.tradukisto.internal.languages.persian.IntegerToWordsConverter integerConverter = 
+                new pl.allegro.finance.tradukisto.internal.languages.persian.IntegerToWordsConverter(
+                hundredsToStringConverter, persianValues.pluralForms());
+
+        return new Container(integerConverter,
+                new pl.allegro.finance.tradukisto.internal.languages.persian.BigDecimalToBankingMoneyConverter(integerConverter,
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/Container.java:173]: please use import and not the full qualified class name

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/BigDecimalToBankingMoneyConverter.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
+package pl.allegro.finance.tradukisto.internal.languages.persian;
+
+import java.math.BigDecimal;
+
+import pl.allegro.finance.tradukisto.internal.BigDecimalToStringConverter;
+import pl.allegro.finance.tradukisto.internal.IntegerToStringConverter;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.String.format;
+
+public class BigDecimalToBankingMoneyConverter implements BigDecimalToStringConverter {
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/BigDecimalToBankingMoneyConverter.java:10]: rename class to `PersianBigDecimalToBankingMoneyConverter`

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/BigDecimalToBankingMoneyConverter.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        this.currencySymbol = currencySymbol;
+    }
+
+    @Override
+    public String asWords(BigDecimal value) {
+        validate(value);
+
+        Integer units = value.intValue();
+        Integer subunits = value.remainder(BigDecimal.ONE).multiply(new BigDecimal(100)).intValue();
+        if(subunits > 0)
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/BigDecimalToBankingMoneyConverter.java:30]: add space after `if`, add braces to `if` and `else` blocks

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+import pl.allegro.finance.tradukisto.internal.GenderAwareIntegerToStringConverter;
+import pl.allegro.finance.tradukisto.internal.IntegerToStringConverter;
+import pl.allegro.finance.tradukisto.internal.languages.PluralForms;
+
+/**
+ * @author omidp
+ *
+ */
+public class IntegerToWordsConverter extends pl.allegro.finance.tradukisto.internal.converters.IntegerToWordsConverter
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:17]: rename class to `PersianIntegerToWordsConverter` and add import for `pl.allegro.finance.tradukisto.internal.converters.IntegerToWordsConverter`

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+{
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:18]: move opening braces (not just this one) to the line above

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        super(hundredsToWordsConverter, pluralForms);
+        this.pluralForms = pluralForms;
+    }
+
+    public IntegerToWordsConverter(IntegerToStringConverter hundredsToWordsConverter, List<PluralForms> pluralForms)
+    {
+        super(hundredsToWordsConverter, pluralForms);
+        this.pluralForms = pluralForms;
+    }
+
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:33]: that's an unnecessary blank line

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+                        String string = (String) iterator.next();
+                        sb.append(string);                        
+                    }
+                }
+                else
+                    sb.append(item).append(" ");
+            }
+            counter++;
+        }
+//        return Joiner.on(" ").join(result).trim();
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:62]: remove comment

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    protected String joinParts(List<String> result)
+    {
+        if (result.size() == 0) {
+            return hundredsToWordsConverter.asWords(0, pluralForms.get(0).genderType());
+        }
+        StringBuilder sb = new StringBuilder();
+        int counter = 0;
+        for (String item : result)
+        {
+            if(item != null && item.trim().length() > 0)
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:45]: space after `if`

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+            {
+                if(counter > 1)
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:47]: space after `if`

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+                {
+                    Iterable<String> split = Splitter.on(" ").split(item);
+                    for (Iterator iterator = split.iterator(); iterator.hasNext();)
+                    {                    
+                        sb.append(" \u0648 ");
+                        String string = (String) iterator.next();
+                        sb.append(string);                        
+                    }
+                }
+                else
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/IntegerToWordsConverter.java:57]: add braces

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/PersianValues.java [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+ * @author omidp
+ *
+ */
+public class PersianValues implements BaseValues
+{
+
+    @Override
+    public Map<Integer, GenderForms> baseNumbers()
+    {
+        return baseNumbersBuilder().put(0, "\u0635\u0641\u0631").put(1, "\u06CC\u06A9").put(2, "\u062F\u0648").put(3, "\u0633\u0647")
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/PersianValues.java:23]: reformat so that each `put` is on a separate line

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/PersianValues.java [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+                .put(100, "\u0635\u062F").put(200, "\u062F\u0648\u06CC\u0633\u062A").put(300, "\u0633\u06CC\u0635\u062F")
+                .put(400, "\u0686\u0647\u0627\u0631\u0635\u062F").put(500, "\u067E\u0627\u0646\u0635\u062F")
+                .put(600, "\u0634\u0634\u0635\u062F").put(700, "\u0647\u0641\u062A\u0635\u062F").put(800, "\u0647\u0634\u062A\u0635\u062F")
+                .put(900, "\u0646\u0647\u0635\u062F").build();
+    }
+
+    @Override
+    public List<PluralForms> pluralForms()
+    {
+        return Arrays.asList(new PersianPluralForms(""), new PersianPluralForms("\u0647\u0632\u0627\u0631"),
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/PersianValues.java:41]: reformat so that each constructor call is on a separate line

File: src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/PersianValues.java [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+                new PersianPluralForms("\u0645\u06CC\u0644\u06CC\u0648\u0646"),
+                new PersianPluralForms("\u0645\u06CC\u0644\u06CC\u0627\u0631\u062F"));
+    }
+
+    @Override
+    public String currency()
+    {
+        return "";
```
Comment (MaciejLipinski) [inline src/main/java/pl/allegro/finance/tradukisto/internal/languages/persian/PersianValues.java:49]: add currency symbol or code

**Review by MaciejLipinski — CHANGES_REQUESTED** [review]
Hello, thank you for your contribution. I recently took over the project and I hope your (well-aged) PR could still be merged! Before that could happen, some things need fixing (see comments), and you would have to add tests – this PR has none. I'll be happy to help if needed.
